enterprise/server/backends/migration_cache/migration_cache.go

```diff
@@ -407,16 +411,19 @@ func (d *doubleReader) Read(p []byte) (n int, err error) {
 	if d.dest != nil {
 		eg.Go(func() error {
 			pCopy := make([]byte, len(p))
-			_, dstErr = d.dest.Read(pCopy)
+			var dstN int
+			dstN, dstErr = d.dest.Read(pCopy)
+			d.bytesReadDest += dstN
```

Comment:
bytesReadDest needs to be protected by a mutex since it's being read/written by multiple goroutines